components/script/serviceworker_manager.rs

```diff
@@ -108,76 +188,194 @@ impl ServiceWorkerManager {
         }
     }
 
-    fn forward_message(&self, msg: DOMMessage, sender: &Sender<ServiceWorkerScriptMsg>) {
-        let DOMMessage { origin, data } = msg;
-        let _ = sender.send(ServiceWorkerScriptMsg::CommonWorker(
-            WorkerScriptMsg::DOMMessage { origin, data },
-        ));
+    fn handle_message_from_resource(&mut self, mediator: CustomResponseMediator) -> bool {
+        if serviceworker_enabled() {
+            if let Some(scope) = self.get_matching_scope(&mediator.load_url) {
+                if let Some(registration) = self.registrations.get_mut(&scope) {
```

Comment:
Looks like we don't need to mutate this registration. Can we just use `get` instead 👀 ?